app/Http/Controllers/Service/Admin/DeliveriesController.php

```diff
@@ -2,32 +2,73 @@
 
 namespace App\Http\Controllers\Service\Admin;
 
+use App\Centre;
 use App\Delivery;
-use App\Sponsor;
 use App\Http\Controllers\Controller;
+use App\Http\Requests\AdminNewDeliveryRequest;
+use App\Sponsor;
+use App\Voucher;
+use App\VoucherState;
+use Auth;
+use Carbon\Carbon;
+use DB;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
+use Illuminate\View\View;
+use Log;
+use Throwable;
 
 class DeliveriesController extends Controller
 {
+    /**
+     * Determines if the given voucher range contains entries already delivered
+     *
+     * @param $start
+     * @param $end
+     * @param $shortcode
+     * @return bool
+     * @throws Throwable
+     */
+    public static function rangeIsUndelivered($start, $end, $shortcode)
+    {
+        $undeliveredRanges = Voucher::getUndeliveredVoucherRangesByShortCode($shortcode);
+
+        // If NOT an empty array
+        if (!empty($undeliveredRanges)) {
```

Comment:
I think this condition may be redundant

Reply:
True! it returns an array, which may be empty, so... we wouldn't iterate over it.